packages/app-store/googlecalendar/lib/CalendarService.ts

```diff
@@ -31,34 +32,89 @@ interface GoogleCalError extends Error {
 
 const ONE_MINUTE_MS = 60 * 1000;
 const CACHING_TIME = ONE_MINUTE_MS;
+const MS_PER_DAY = 24 * 60 * 60 * 1000;
+const ONE_MONTH_IN_MS = 30 * MS_PER_DAY;
 
 /** Expand the start date to the start of the month */
 function getTimeMin(timeMin: string) {
+  const date = new Date();
+  const dateMonth = date.getMonth();
   const dateMin = new Date(timeMin);
+  const dateMinMonth = dateMin.getMonth();
+  // If we browser the next month, return the start of the previous month to guarantee a cache hit
+  if (date.getFullYear() === dateMin.getFullYear() && dateMinMonth - dateMonth === 1) {
+    return new Date(date.getFullYear(), date.getMonth(), 1, 0, 0, 0, 0).toISOString();
+  }
   return new Date(dateMin.getFullYear(), dateMin.getMonth(), 1, 0, 0, 0, 0).toISOString();
 }
 
 /** Expand the end date to the end of the month */
 function getTimeMax(timeMax: string) {
+  const date = new Date();
+  const dateMonth = date.getMonth();
   const dateMax = new Date(timeMax);
-  return new Date(dateMax.getFullYear(), dateMax.getMonth() + 1, 0, 0, 0, 0, 0).toISOString();
+  const dateMaxMonth = dateMax.getMonth();
+  if (
+    // Normalize the next two months requests to guarantee a cache hit
+    date.getFullYear() === dateMax.getFullYear() &&
+    (dateMaxMonth - dateMonth === 1 || dateMaxMonth - dateMonth === 2)
+  ) {
+    return new Date(date.getFullYear(), date.getMonth() + 2, 1, 0, 0, 0, 0).toISOString();
+  }
+  return new Date(dateMax.getFullYear(), dateMax.getMonth(), 1, 0, 0, 0, 0).toISOString();
 }
```

Comment:
`suggestion`: For readability and maintainability:

```suggestion
function getTimeMax(timeMax: string): string {
  const currentDate = new Date();
  const currentMonth = currentDate.getMonth();
  const targetDate = new Date(timeMax);
  const targetMonth = targetDate.getMonth();

  const isSameYear = currentDate.getFullYear() === targetDate.getFullYear();
  const isNextTwoMonths = [1, 2].includes(targetMonth - currentMonth);

  if (isSameYear && isNextTwoMonths) {
    return formatDate(currentDate.getFullYear(), currentMonth + 2);
  }

  return formatDate(targetDate.getFullYear(), targetMonth);
}

function formatDate(year: number, month: number): string {
  return new Date(year, month, 1, 0, 0, 0, 0).toISOString();
}
```